_config/email.yml

```diff
@@ -1,12 +1,55 @@
 ---
-Name: emailconfig
+Name: mailer
 ---
 SilverStripe\Core\Injector\Injector:
-  Swift_Transport: Swift_MailTransport
-  Swift_Mailer:
-    constructor:
-      - '%$Swift_Transport'
-  SilverStripe\Control\Email\Mailer:
-    class: SilverStripe\Control\Email\SwiftMailer
-    properties:
-      SwiftMailer: '%$Swift_Mailer'
+  Symfony\Component\Mailer\MailerInterface:
+    class: Symfony\Component\Mailer\Mailer
```

Comment:
We're using still using original Symfony Mailer, which is a final class

Another option was to create own class that implemented MailerInterface, however I think it's better to be more strict and use the official Symfony class as the foundation here, as it's much more likely to "just work" when using services beyond just regular sendmail e.g. email being sent on an async queue

We used to customise the headers on the email sent with a SwiftPlugin - https://github.com/silverstripe/silverstripe-framework/blob/4/src/Control/Email/SwiftPlugin.php - however the "plugin" architecture no longer exists

To replicate what the the plugin used to do, we now use a new customer class MailerSubscriber which uses the Symfony EventDispatcher architecture

However in order to use the EventDispatcher, Mailer demands that we also supply a MessageBus

The MessageBus itself is pretty complicated to create and is largely responsible for the somewhat complicated yml here

However in the end, despite the complication here, I think this is good as it makes it easier for anyone else to copy paste this yml into their own project and use it as a basis for setting up the various symfony classes to support complex email setups